met-web/src/components/engagement/form/EngagementWidgets/MetWidget.tsx

```diff
@@ -50,14 +50,16 @@ const MetWidget = ({
                 </Grid>
                 <Grid item xs={2} container direction="row" alignItems="flex-start" justifyContent="center">
                     <Stack direction="row" spacing={1}>
-                        <IconButton
-                            sx={{ margin: 0, padding: 0 }}
-                            color="inherit"
-                            onClick={onEdit}
-                            data-testid="widget/edit"
-                        >
-                            <EditIcon />
-                        </IconButton>
+                        <When condition={typeof onEdit === 'function'}>
```

Comment:
in which cases is it not a type of function?

Reply:
i think you need a new prop David, something like "editable" that should be passed as false for this widget

Reply:
look at the optional "sortable" prop

Reply:
Good suggestions, Ill change it to the !!onEdit conditional so we can prevent adding more props